### PR TITLE
fix(#8886): block users from editing important fields in their settings

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

#8886 

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs
If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8886-block-user-settings-updates/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8886-block-user-settings-updates/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8886-block-user-settings-updates/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: admin/src/js/controllers/display-privacy-policies-preview.js [hunk 1]
```diff
@@ -5,14 +5,20 @@ angular
...
     $timeout,
-    $uibModalInstance,
-    PrivacyPolicies
+    $uibModalInstance
   ) {
     'use strict';
     'ngInject';
 
     const getHtml = string => {
-      return $sanitize(PrivacyPolicies.decodeUnicode(string));
```
Comment (garethbowen) [inline admin/src/js/controllers/display-privacy-policies-preview.js:15]: This is a refactor that I found while working out the privacy policy fields in user settings - by inlining this one function we can drop the entire service.

File: ddocs/medic-db/medic-client/validate_doc_update.js [hunk 2]
```diff
@@ -13,11 +13,42 @@ function(newDoc, oldDoc, userCtx) {
     throw({ forbidden: msg });
   };
 
+  var hasRole = function(roles, role) {
```
Comment (garethbowen) [inline ddocs/medic-db/medic-client/validate_doc_update.js:16]: Copied from the server validate_doc_update function. We don't yet have a sensible way to include functions like this.

File: ddocs/medic-db/medic/validate_doc_update.js [hunk 3]
```diff
@@ -37,18 +37,17 @@ function(newDoc, oldDoc, userCtx, secObj) {
...
-    if (secObj.admins && secObj.admins.names &&
-        secObj.admins.names.indexOf(userCtx.name) !== -1) {
+    if (secObj.admins && secObj.admins.names && secObj.admins.names.indexOf(userCtx.name) !== -1) {
       return true;
     }
 
     if (secObj.admins && secObj.admins.roles) {
-      for (var i = 0; i < userCtx.roles; i++) {
-        if (hasRole(secObj.admins, userCtx.roles[i])) {
+      for (var i = 0; i < userCtx.roles.length; i++) {
```
Comment (garethbowen) [inline ddocs/medic-db/medic/validate_doc_update.js:49]: There was a bug here - without the `.length` this loop was never executing so national_admins were blocked. Found when adding unit tests for this PR.

**Review by dianabarsan — APPROVED** [review]
Sweet!

**Pull request merged**